packages/axios/src/index.ts

```diff
@@ -1,52 +1,172 @@
 import http from 'node:http';
 import https from 'node:https';
 
-import type { IConfig } from '@cortec/config';
-import type { IContext, IModule } from '@cortec/types';
-import type { Axios, AxiosRequestConfig } from 'axios';
-import axios from 'axios';
+import { type IConfig, Config } from '@cortec/config';
+import type { INewrelic } from '@cortec/newrelic';
+import {
+  type IContext,
+  type IModule,
+  type ITrace,
+  Headers,
+} from '@cortec/types';
+import type * as A from 'axios';
+import axios, { isAxiosError } from 'axios';
+import capitalize from 'lodash.capitalize';
 
 export interface IAxiosConfig {
-  global?: AxiosRequestConfig;
+  global?: A.AxiosRequestConfig;
   api: {
-    [key: string]: AxiosRequestConfig;
+    [key: string]: A.AxiosRequestConfig;
   };
 }
 
-export interface IRequester {
-  api(name: string): Axios;
+interface TracedAxios extends A.Axios {
+  trace(ctx: ITrace): A.Axios;
 }
 
-export default class CortecAxios implements IModule, IRequester {
+export interface IAxios {
+  service(name: string, flags?: RequestFlags): TracedAxios;
+}
+
+export enum RequestFlags {
+  None = 0,
+  Notice4XX = 1 << 0,
```

Comment:
Why the nop shift? Just for consistency if other flags are added later?

Reply:
Yup.